File: ex_happy-primes/submissions/kevin-php/happyprime.php
<?$t=$argv[1];for($z=$t-1,$p=1;$z&&++$p<$t;)$z=$t%$p;$f=array(1);while(!in_array($t,$f,1)){$f[]=$t;$t=array_reduce(str_split($t),function($v,$c){return $v+=$c*$c;});}print($t<2?"happy ":"sad ").(!$z?"non-":"")."prime";?>

File: ex_happy-primes/reference/happyprime.py
import sys
n,j=int(sys.argv[1]),'prime'
k='non-'+j
def h(n):
 s=set()
 while n!=1:
  n=sum(int(a)**2 for a in str(n))
  if n in s:return'sad'
  s.add(n)
 return'happy'
def p(n):
 if n==1:return k
 for x in xrange(2,int(n**0.5)+1):
  if n%x==0:return k
 return j
print h(n),p(n)
